src/js/components/List/List.js

```diff
@@ -0,0 +1,232 @@
+import React from 'react';
+import { compose } from 'recompose';
+import styled, { css, withTheme } from 'styled-components';
+
+import { Box } from '../Box';
+import { Button } from '../Button';
+import { InfiniteScroll } from '../InfiniteScroll';
+import { Keyboard } from '../Keyboard';
+import { Text } from '../Text';
+import { backgroundStyle, focusStyle, normalizeColor } from '../../utils';
+import { withFocus, withForwardRef } from '../hocs';
+
+const StyledList = styled(Box)`
+  list-style: none;
+  margin: 0;
+  padding: 0;
+  ${props => props.focus && focusStyle}
+`;
+
+const hoverStyle = css`
+  ${props =>
+    backgroundStyle(
+      normalizeColor(props.theme.global.hover.background, props.theme),
+      props.theme,
+    )}
+  color: ${props =>
+    normalizeColor(props.theme.global.hover.color, props.theme)};
+`;
+
+const StyledButton = styled(Button)`
+  ${props => props.active && hoverStyle}
+`;
+
+const normalize = (item, index, property) => {
+  if (typeof property === 'function') {
+    return property(item, index);
+  }
+  return item[property];
+};
+
+const List = React.forwardRef((props, ref) => {
+  const {
+    action,
+    background,
+    border,
+    children,
+    data,
+    focus,
+    itemProps,
+    pad,
+    primaryKey,
+    secondaryKey,
+    step,
+    theme,
+    onClickItem,
+    onMore,
+    ...rest
+  } = props;
+  const [active, setActive] = React.useState();
+
+  return (
+    <Keyboard
+      onEnter={
+        onClickItem && active >= 0
+          ? event => {
+              event.persist();
+              const adjustedEvent = event;
+              adjustedEvent.item = data[active];
+              adjustedEvent.index = active;
+              onClickItem(adjustedEvent);
+            }
+          : undefined
+      }
+      onUp={
+        onClickItem && active
+          ? () => {
+              setActive(active - 1);
+            }
+          : undefined
+      }
+      onDown={
+        onClickItem && data.length
+          ? () => {
+              setActive(
+                active >= 0 ? Math.min(active + 1, data.length - 1) : 0,
+              );
+            }
+          : undefined
+      }
+    >
+      <StyledList
+        as="ol"
```

Comment:
does it mean we only support an ordered list?

Reply:
Added `as` to List.